crates/ruff/resources/test/fixtures/pylint/repeated_equality_comparison_target.py

```diff
@@ -9,13 +9,20 @@
 
 foo == a or foo == "b" or foo == 3  # Mixed types.
 
-# False negatives (the current implementation doesn't support Yoda conditions).
 "a" == foo or "b" == foo or "c" == foo
 
 "a" != foo and "b" != foo and "c" != foo
 
 "a" == foo or foo == "b" or "c" == foo
 
+foo == bar or baz == foo or qux == foo
+
+foo == "a" or "b" == foo or foo == "c"
+
+foo != "a" and "b" != foo and foo != "c"
+
+foo == "a" or foo == "b" or "c" == bar or "d" == bar  # Multiple targets
```

Comment:
We have odd behavior for this kind of thing on `main` already:

```
         49 │+repeated_equality_comparison_target.py:40:1: PLR1714 Consider merging multiple comparisons: `foo in ("a", "b")`. Use a `set` if the elements are hashable.
         50 │+   |
         51 │+38 | foo == bar == "b" or foo == "c"  # Multiple comparisons.
         52 │+39 |
         53 │+40 | foo == "a" or foo == "b" or bar == "c" or bar == "d"  # Multiple targets
         54 │+   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ PLR1714
         55 │+   |
         56 │+
         57 │+repeated_equality_comparison_target.py:40:1: PLR1714 Consider merging multiple comparisons: `bar in ("c", "d")`. Use a `set` if the elements are hashable.
         58 │+   |
         59 │+38 | foo == bar == "b" or foo == "c"  # Multiple comparisons.
         60 │+39 |
         61 │+40 | foo == "a" or foo == "b" or bar == "c" or bar == "d"  # Multiple targets
         62 │+   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ PLR1714
```

It's not _awful_ because it's not like we autofix this and remove the comparisons, but it'd be nice to have a more targeted range. Anyway...